destroy.sh

```diff
@@ -106,26 +105,63 @@ do
   fi
 
   # Empty the bucket
+  echo "emptying bucket: $i"
   aws s3 rm s3://$i/ --recursive
 done
 
-echo "Removing certificate from stage"
-
 restApiName=$stage-app-api
 
 restApiId=$(aws apigateway get-rest-apis | jq -r ".[] | .[] |  select(.name==\"$restApiName\") | .id|tostring")
 
-aws apigateway update-stage \
-  --rest-api-id $restApiId \
-  --stage-name $stage \
-  --patch-operations op=replace,path=/clientCertificateId,value="" \
-  &>/dev/null
+if [ "$restApiId" != "" ]; then
+  echo "Removing certificate from stage"
+
+  aws apigateway update-stage \
+    --rest-api-id $restApiId \
+    --stage-name $stage \
+    --patch-operations op=replace,path=/clientCertificateId,value="" \
+    &>/dev/null
+
+  restApiArn="arn:aws:apigateway:us-east-1::/restapis/$restApiId/stages/$stage"
 
-echo "Removed certificate from stage"
+  echo "Disassociating web acl from api gateway"
+
+  aws wafv2 disassociate-web-acl \
+    --resource-arn $restApiArn &>/dev/null
+
+  echo "Removed certificate from stage and disassociated web acl"
+fi
 
 # Trigger a delete for each cloudformation stack
 for i in "${stackList[@]}"
 do
-  echo $i
+  echo "triggering stack deletion for $i"
   aws cloudformation delete-stack --stack-name $i
 done
+
+# Delete Client Certificates associated with a branch
+certToDestroy=$(aws apigateway get-client-certificates | grep \"app-api-${stage}\" -B 2 |
+  grep -o '"clientCertificateId": "[^"]*' |
+  grep -o '[^"]*$')
+
+until [ -z $certToDestroy ]; do
+  aws apigateway delete-client-certificate --client-certificate-id $certToDestroy || true
+  sleep 10
+  certToDestroy=$(aws apigateway get-client-certificates | grep \"app-api-${stage}\" -B 2 |
+    grep -o '"clientCertificateId": "[^"]*' |
+    grep -o '[^"]*$' || true)
+done
```

Comment:
copied from MCR – deletes client cert to make sure we don't hit our limit
(can verify by going to api gateway -> select any api -> client certificates -> make sure the one with your branch name disappears at the end)